recipes/faiss/build-lib.sh

```diff
@@ -0,0 +1,17 @@
+CUDA_CONFIG_ARG=""
+if [ ${cuda_compiler_version} != "None" ]; then
+    CUDA_CONFIG_ARG="--with-cuda=${CUDA_HOME}"
+else
+    CUDA_CONFIG_ARG="--without-cuda"
+fi
+
+# Build vanilla version (no avx)
```

Comment:
Commenting here, but this applies to other builds in this PR as well.

Would it make sense to build the AVX bits? IIUC our compilers build with a somewhat recent/common chip target while including a fallback path for systems that don't support it (though others please correct me if this is wrong/out-of-date).

Reply:
I was wondering about this as well, but this seems like a more high-level discussion, and maybe something that should be taken care of by a virtual package? Not sure if CF wants to assume that all x86_64 installs have avx2 available? This is gonna become relevant fairly soon IMO (cf. https://github.com/numpy/numpy/pull/13516), but it wasn't something I wanted to get into for this PR.

Reply:
Maybe @isuruf or @beckermr can fill in more details here.

When I last looked (which was not recent btw), we would target Haswell and fallback to Nocona. So it would try to use AVX2, but degrade gracefully if your architecture is older than that. Admittedly things go slower if you don't have AVX support, but it seems like an acceptable compromise for an older CPU (and better than not running at all). The exact chips may be different today, but the general behavior I believe is the same. Though am having trouble finding exactly where this is specified atm.

Reply:
That doesn't make sense. You can't target haswell and fallback to nocona unless the software itself supports it. (gcc has a -mtune parameter that we turn on, but that doesn't help much and not at all in most cases.)

Reply:
There are 3 levels of optimization in Faiss x64:

- nothing --> uses only scalar ops

- `-msse4` --> SSE4 codepaths

- `-mavx2` --> best optimization

Also it is quite important to compile with `-mpopcnt` because the binary indexes rely heavily on it. In all cases, the code compiles (the popcount instruction is handled by the `__builtin_popcountl` GCC intrinsic), but the performance impact can be serious.